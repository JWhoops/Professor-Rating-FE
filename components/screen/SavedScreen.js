import * as React from 'react';
import { ActivityIndicator, IconButton, Button } from 'react-native-paper';
import { View, FlatList, Text } from 'react-native';
import ResultSaved from '../ResultSaved.js'; //


class SavedScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      url: 'http://172.220.7.76:8080',
      courses: '',
      search: '',
      terms: [],
      fav: [],
      visible: true,
      token: '',
      notFound: false,
    };
    this.updateList = this.updateList.bind(this);
  }

  static navigationOptions = (
    { navigation }) => {
    return {
      title: '',
      headerStyle: {
        backgroundColor: '#C5050C',
        shadowOffset: {
          // 设置阴影偏移量
          width: 0,
          height: 4,
        },
        shadowRadius: 10, // 设置阴影模糊半径
        shadowOpacity: 0.0, // 设置阴影的不透明度
        shadowColor: 'rgba(96,96,96,1)', // 设置阴影色
        elevation: 4,
      },
      headerLeft: (() =>
        <IconButton
          accessibilityLabel={'saved_back_btn'}
          testID={'saved_back_btn'}
          accessible={true}
          icon={'chevron-left'}
          onPress={() => { navigation.goBack() }}
          color="#F7F7F7"
          size={35} />),
    }
  }

  updateList() {
    const { url, email, token } = this.props.navigation.state.params;
    fetch(url + '/v1/users/email/' + email, {
      method: 'GET',
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
        Accept: '*/*'
      },
    })
      .then(response => {
        return response.json();
      })
      .then(result => {
        if (result.success === true) {
          // let terms = result.data;
          let terms = result.data.favorite;
          let notFound = false;
          if (terms.length === 0) {
            notFound = true;
          }
          this.setState({ notFound: notFound, terms: terms, visible: false, token: token });
        }
      })
      .catch(err => {
        alert(err);
      });
  }

  componentDidMount() {
    const { url, email, token } = this.props.navigation.state.params;
    fetch(url + '/v1/users/email/' + email, {
      method: 'GET',
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
        Accept: '*/*'
      },
    })
      .then(response => {
        return response.json();
      })
      .then(result => {
        if (result.success === true) {
          // let terms = result.data;
          let terms = result.data.favorite;
          let notFound = false;
          if (terms.length === 0) {
            notFound = true;
          }
          this.setState({ notFound: notFound, terms: terms, visible: false, token: token });
        }
      })
      .catch(err => {
        alert(err);
      });
  }

  render() {
    return (
      <View >
        {this.state.notFound &&
          <View>
            <Text style={{ fontSize: 18, color: "grey", fontWeight: "bold", textAlign: "center", width: "100%", marginTop: "50%" }}
              accessibilityLabel={'no_favorite_text'}
              testID={'no_favorite_text'}
              accessible={true}
            >
              Nothing Here, Bookmark a course : )
            </Text>
            <Button
              accessibilityLabel={'no_favorite_btn'}
              testID={'no_favorite_btn'}
              accessible={true}
              color={"#c5050c"}
              style={{ fontSize: 23, width: "auto" }}
              onPress={() => { this.props.navigation.goBack(); }}
            >Go Back</Button>
          </View>
        }
        {this.state.visible && <ActivityIndicator style={{
          position: "absolute",
          marginTop: "40%",
          alignSelf: "center"
        }} size={60} color="#c5050c" />}
        <FlatList
          data={this.state.terms}
          renderItem={({ item }) => (
            <ResultSaved
              title={item.name}
              course_id={item.id}
              description={item.description}
              navigate={this.props.navigation.navigate}
              token={this.state.token}
              url={this.state.url}
              email={this.state.email}
              id={this.state.id}
              updateList={this.updateList}
            />
          )}
          keyExtractor={item => item.id}
        />
      </View>
    );
  }
}




export default SavedScreen;