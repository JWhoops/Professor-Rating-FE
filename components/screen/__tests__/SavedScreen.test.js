import React from 'react';
import { render } from 'react-native-testing-library';

import SavedScreen from '../SavedScreen';

test('SavedScreen renders all inputs as expected', () => {
    global.fetch = require('node-fetch');
  const createTestProps = (props) => ({
    navigation: {
      state: { params: {
        url: 'http://172.220.7.76:8080', 
        email: 'example@yahoo.com',
      } },
      navigate: jest.fn(),
    },
    ...props
  });

  let props = createTestProps();
  const { toJSON } = render(<SavedScreen {...props}/>);

  expect(toJSON()).toMatchSnapshot();
});