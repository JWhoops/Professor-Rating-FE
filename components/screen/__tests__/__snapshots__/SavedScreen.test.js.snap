// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedScreen renders all inputs as expected 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "alignSelf": "center",
          "marginTop": "40%",
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "height": 60,
          "opacity": 1,
          "width": 60,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 60,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 30,
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 60,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 60,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 30,
                    "overflow": "hidden",
                    "width": 60,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#0000ff",
                      "borderRadius": 30,
                      "borderWidth": 6,
                      "height": 60,
                      "width": 60,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 60,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 30,
                "overflow": "hidden",
                "top": 30,
                "width": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 60,
                  "transform": Array [
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 60,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 30,
                    "overflow": "hidden",
                    "width": 60,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#0000ff",
                      "borderRadius": 30,
                      "borderWidth": 6,
                      "height": 60,
                      "width": 60,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
